internal/security/secretstore/init.go

```diff
@@ -109,20 +123,33 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, _ *sync.WaitGroup, _ s
 					initResponse.RootToken = ""
 					lc.Info("Root token stripped from init response for security reasons")
 				}
-				if err := saveInitResponse(lc, fileOpener, configuration.SecretService, &initResponse); err != nil {
-					lc.Error(fmt.Sprintf("unable to save init response: %s", err.Error()))
-					os.Exit(1)
-				}
 				_, err = vc.Unseal(&initResponse)
 				if err == nil {
 					shouldContinue = false
 				}
+				// We need the unencrypted initResponse in order to generate a temporary root token later
+				// Make a copy and save the copy, possibly encrypted
+				encryptedInitResponse := initResponse
+				// Optionally encrypt the vault init response based on whether encryption was enabled
+				if err := vmkEncryption.EncryptInitResponse(&encryptedInitResponse); err != nil {
```

Comment:
I like methods that maybe do nothing to be called `MaybeEncryptInitResponse`. Mild preference to re-name, OTOH the other pattern I've seen would be to have `vmkEncryption.LoadIKM()` return a specific sentinel error value that you can check for and if that is returned then the caller doesn't have hardware encryption support and thus doesn't ever call EncryptInitResponse, and EncryptInitResponse will be a bit simpler and easier to reason about since it doesn't have this "maybe" quality about it.

Reply:
Resolved by "if vmkEncryption.IsEncrypting()" guard around the call and throwing an error inside of the implementation is a key was not supplied.